sympy/ntheory/factor_.py

```diff
@@ -520,11 +520,19 @@ def perfect_power(n, candidates=None, big=True, factor=True):
 
     n = as_int(n)
     if n < 0:
-        pp = perfect_power(-n)
+        pp = perfect_power(-n, candidates=candidates, big=big, factor=factor)
         if pp:
             b, e = pp
-            if e % 2:
-                return -b, e
+            odd_divisors = []
+            for i in range(3, e + 1, 2):
+                if e % i == 0 and i > 1:
+                    odd_divisors.append(i)
+
+            if odd_divisors:
+                chosen_factor = max(odd_divisors) if big else min(odd_divisors)
+                power = e // chosen_factor
+                new_base = -int(b ** power)
+                return new_base, chosen_factor
```

Comment:
pp = perfect_power(-n, candidates, big=True, factor=factor)
        if pp:
            b, e=pp
            e2 = e&(-e)
            e //= e2
            b = b**e2
            if not big and not isprime(e):
                for small in primerange(3, e+1):
                    if not e % small:
                        break
                b = b**(e//small)
                e = small
            return -b, e